sources/packages/api/src/services/student/student.service.ts

```diff
@@ -29,7 +29,7 @@ export class StudentService extends RecordDataModelService<Student> {
       .createQueryBuilder("student")
       .leftJoinAndSelect("student.user", "user")
       .where("user.userName = :userNameParam", { userNameParam: userName })
-      .getOneOrFail();
+      .getOne();
```

Comment:
can you please explain, why did we change from getOneOrFail to getOne?

Reply:
This was causing an "Internal Server Error" from the API. All places calling this method were testing if the result was null, which never happens because an exception was thrown because of the "OrFail".
Changing it allows the API to return a nice "404 not found" with the proper message.